### PR TITLE
Validate dict when parsing the mkdocs.yml file

### PR DESCRIPTION
**Pull request by saadmk11**
closes: #5481

### CODE REVIEW

File: readthedocs/doc_builder/exceptions.py [hunk 1]
```diff
@@ -76,3 +76,11 @@ class MkDocsYAMLParseError(BuildEnvironmentError):
         'The "{config}" config from your MkDocs YAML config file has to be a '
         'a list of relative paths.',
     )
+
+    EMPTY_CONFIG = 'Please make sure the MkDocs YAML configuration file is not empty.'
```
Comment (stsewd) [inline readthedocs/doc_builder/exceptions.py]: Missing the translation thing `ugettext_noop`

File: readthedocs/doc_builder/exceptions.py [hunk 2]
```diff
@@ -76,3 +76,11 @@ class MkDocsYAMLParseError(BuildEnvironmentError):
...
+
+    CONFIG_NOT_DICT = ugettext_noop(
+        'Your MkDocs YAML configuration was incorrect. '
```
Comment (stsewd) [inline readthedocs/doc_builder/exceptions.py]:
```suggestion
        'Your MkDocs YAML config file is incorrect. '
```

File: readthedocs/doc_builder/backends/mkdocs.py [hunk 3]
```diff
@@ -84,7 +84,18 @@ def load_yaml_config(self):
         :raises: ``MkDocsYAMLParseError`` if failed due to syntax errors.
         """
         try:
-            return yaml.safe_load(open(self.yaml_file, 'r'),)
+            config = yaml.safe_load(open(self.yaml_file, 'r'),)
```
Comment (stsewd) [inline readthedocs/doc_builder/backends/mkdocs.py]: We can remove that trailing comma

```suggestion
            config = yaml.safe_load(open(self.yaml_file, 'r'))
```

**Comment by saadmk11** [comment]
@stsewd Updated the PR

**Comment by ericholscher** [comment]
🎉

**Pull request merged**